Start explaining parameters (#80)
Paperback-iOS/website: src/help/guides/extension-development/function-definitions.md

# Required Methods
## getMangaDetails
### Method signature
`async getMangaDetails(mangaId: string): Promise<Manga>`

### Parameters
| Parameter | Type | Description|
|-----------|------|------------|

> |  mangaId  | String | The ID of a manga. The manga ID is provided from the other discovery functions, such as the home page and performed searches. The manga ID should be used to return information about the manga, such as the title and author.|

Reviewer: What would you think of using a code bloc? I'm not sure if its better with or without
```suggestion
|  `mangaId`  | String | The ID of a manga. The manga ID is provided from the other discovery functions, such as the home page and performed searches. The manga ID should be used to return information about the manga, such as the title and author.|
```

Author: Should this be done for every parameter? I mean it would help distinguish between uppercase I's and lowercase l's.

### Returns

> Given an ID, this async function should return a filled out [Manga](model-reference.md#Manga) object.

Reviewer: We usually don't use file extension in links. Should we?
```suggestion
Given an ID, this async function should return a filled out [Manga](model-reference/#Manga) object.
```

Author: It doesn't make a difference from what I've seen.

# Required Methods

## getMangaDetails

### Method signature

`async getMangaDetails(mangaId: string): Promise<Manga>`

### Parameters

| Parameter | Type | Description|
|-----------|------|------------|

> |  `mangaId`  | String | The ID of a manga. The manga ID is provided from the other discovery functions, such as the home page and performed searches. The manga ID should be used to return information about the manga, such as the title and author.|

Reviewer 2: Be specific with the function names. Even better, link them!

Author: The functions aren't added, but I will link them once I do add them in.

```

## getChapterDetails
### Method Signature
`async getChapterDetails(mangaId: string, chapterId: string): Promise<ChapterDetails>`

### Parameters
| Parameter | Type | Description|
|-----------|------|------------|

> |  mangaId  | String | The ID of a manga. The manga ID is provided from the other discovery functions, such as the home page and performed searches. The manga ID should be used to return information about the manga, such as the title and author.|

Reviewer: ```suggestion
|  mangaId  | String | The ID of a manga. The manga ID is provided from the other discovery functions, such as the home page and performed searches. The manga ID should be used to return information about the manga, such as the title and author. |
```

    }

    // Return a list of all of the chapters discovered
    return chapters;
}
```

## getChapterDetails

### Method Signature

`async getChapterDetails(mangaId: string, chapterId: string): Promise<ChapterDetails>`

### Parameters

| Parameter | Type | Description|
|-----------|------|------------|
|  `mangaId`  | String | The ID of a manga. The manga ID is provided from the other discovery functions, such as the home page and performed searches. The manga ID should be used to return information about the manga, such as the title and author. |
| `chapterId` | String | The ID of a chapter. defined by the [Get Chapters](#getchapters) method. The chapter ID can only contain letters, numbers, dashes and underscores. |

### Returns

> Given both a Manga and a Chapter ID, this function should populate a [ChapterDetails](model-reference.md#ChapterDetails)

Reviewer: As you said there was no difference, I would prefer:
```suggestion
Given both a Manga and a Chapter ID, this function should populate a [ChapterDetails](model-reference/#ChapterDetails)
```

Author: Actually, while I did say there was no difference, the latter breaks on my local testing system, so I actually think the first one is the way to go.

Reviewer 2: I'm having better luck with what @FramboisePi has suggested. How does this break your local system?
I'm getting a strange navigation frame center with what's currently present here.

Author: For some reason, it's working now. I may have made a typo earlier causing it to not work, but regardless it seems to work now.

}
```

## searchRequest

### Method signature

`async searchRequest(query: SearchRequest, metadata: any): Promise<PagedResults>`

### Parameters

| Parameter | Type | Description|
|-----------|------|------------|

> | `query` | [SearchRequest](model-reference.md#searchrequest) | The object representing the search. View the fields of the [SearchRequest](model-reference.md#searchrequest) object to find out what a search can contain. |

Reviewer: ```suggestion
| `query` | [SearchRequest](model-reference/#searchrequest) | The object representing the search. View the fields of the [SearchRequest](model-reference/#searchrequest) object to find out what a search can contain. |
```

Author: per above

> | `metadata` | any | Metadata carried through function calls. View the [metadata page](metadata.md) to learn more about how the parameter works. |

Reviewer: ```suggestion
| `metadata` | any | Metadata carried through function calls. View the [metadata page](metadata/) to learn more about how the parameter works. |
```

Author: Per above

### Returns

> A [PagedResults](model-reference.md#pagedresults) object with the results of the search for the current page, and metadata for the next page, if it exists.

Reviewer: ```suggestion
A [PagedResults](model-reference/#pagedresults) object with the results of the search for the current page, and metadata for the next page, if it exists.
```

Author: Per above

### Example Implementation

```ts
async searchRequest(query:SearchRequest, metadata:any): Promise<PagedResults> {
	let page;

	// Get the current page the function is on
	if (typeof metadata === "object" && metadata.page) {
		page = metadata.page;
	} else { // If there is no current page, this must mean this is the first page.
		page = 1;
	}

> 	const request = createRequest({
> 		url: `https://mysite.com/search/?query=${query.title}&page=${page}`,
> 		method: GET,
> 		...
> 	});

Reviewer: Should it be `createRequestObject`?
```suggestion
	const request = createRequestObject({
		url: `https://mysite.com/search/?query=${query.title}&page=${page}`,
		method: GET,
		...
	});
```

	/*
	Execute the request, and retrieve the HTML page
	this.requestManager is provided to you by the parent class. Always use this to make requests.
	The second parameter is the number of retries the app is allowed to make, if the request fails the first time.
	*/
	const data = await this.requestManager.schedule(request, 1)

> 	// Prepare to parse the page using CheerioJS (Class object included by parent class)

Reviewer 2: ```suggestion
	// Prepare to parse the page using CheerioJS (The cheerio object is provided by the base class this extends)
```
I think this was a copy and paste from my original draft, lets do a slight update
